specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2022-11-30-preview/definitions/MigrateMySqlAzureDbForMySqlOfflineTask.json

```diff
@@ -0,0 +1,500 @@
+          "default": false
+        },
+        "migrateAllRoutines": {
+          "type": "boolean",
+          "description": "If true, all routine definitions will be migrated in the selected databases",
+          "default": false
+        },
+        "migrateAllTablesSchema": {
+          "type": "boolean",
+          "description": "If true, all table's schemas will be migrated",
+          "default": false
+        },
+        "migrateUserSystemTables": {
+          "type": "boolean",
+          "description": "If true, all users/grants will be migrated",
+          "default": false
+        },
+        "encryptedKeyForSecureFields": {
+          "type": "string",
+          "description": "encrypted key for secure fields"
+        }
+      },
+      "required": [
+        "sourceConnectionInfo",
+        "targetConnectionInfo",
+        "selectedDatabases"
+      ]
+    },
+    "MigrateMySqlAzureDbForMySqlOfflineTaskOutput": {
+      "type": "object",
+      "description": "Output for the task that migrates MySQL databases to Azure Database for MySQL for offline migrations",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Result identifier",
+          "readOnly": true
+        },
+        "resultType": {
+          "description": "Result type",
+          "type": "string"
+        }
+      },
+      "required": [
+        "resultType"
+      ],
+      "discriminator": "resultType"
+    },
+    "MigrateMySqlAzureDbForMySqlOfflineTaskOutputMigrationLevel": {
+      "x-ms-discriminator-value": "MigrationLevelOutput",
+      "type": "object",
+      "properties": {
+        "migrationType": {
+          "enum": [
+            "InitialLoad"
+          ],
+          "type": "string",
+          "readOnly": true
+        },
+        "startedOn": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Migration start time",
+          "readOnly": true
+        },
+        "endedOn": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Migration end time",
+          "readOnly": true
+        },
+        "durationInSeconds": {
+          "type": "integer",
+          "format": "int64",
+          "description": "Duration of task execution in seconds.",
+          "readOnly": true
+        },
+        "status": {
+          "$ref": "./TasksCommon.json#/definitions/MigrationStatus",
+          "description": "Current status of migration",
+          "readOnly": true
+        },
+        "statusMessage": {
+          "type": "string",
+          "description": "Migration status message",
+          "readOnly": true
+        },
+        "message": {
+          "type": "string",
+          "description": "Migration progress message",
+          "readOnly": true
+        },
+        "databases": {
+          "type": "string",
+          "description": "Selected databases as a map from database name to database id",
```

Comment:
readonly?

Reply:
I had to remove "readonly" from this attribute because it was causing a schema validation error. 

According to the [OpenAPI spec](https://swagger.io/specification/v2/), the "properties marked as readOnly being true SHOULD NOT be in the required list of the defined schema". Unfortunately, it has to be in the required list because it is also a discriminator.